kotest-property/src/commonMain/kotlin/io/kotest/property/arbitrary/collections.kt

```diff
@@ -40,17 +42,24 @@ fun <T> Arb.Companion.of(vararg collection: T): Arb<T> = element(*collection)
 fun <A> Arb.Companion.set(gen: Gen<A>, range: IntRange = 0..100): Arb<Set<A>> {
    check(!range.isEmpty())
    check(range.first >= 0)
-   val maxIterations = 1000
+   // we may generate duplicates, but we don't know if the underlying gen has sufficient cardinality
+   // to satisfy our range, so we can try for a while, but must not try for ever
+   // the slippage factor controls how many times we will accept a non unique element before giving up,
+   // which is the number of elements in the target set * slippage
+   val slippage = 10
```

Comment:
that's a good idea

Reply:
that would probably mean that it's going to yield a very large number of iterations before it gives up when the set size is so big 🤔

Reply:
Yeah, I think because I wasn't resetting it. I was kind of thinking, if you ask for k then 10 x k "misses" is a good value.

Reply:
We maybe can use lower values if we're resetting the iterations on successful sample.

Reply:
I guess it can't be static - if you ask for 10,000 elements, and your arb produces 20,000 then you probably want to run for more than 20k, as 20k will likely have repeats. That's why I settled on an order of magnitude increase. It's not perfect, and could be configurable on another PR ?

Reply:
Because we don't know how many elements the arb can provide, so we're in the blind as to how many times to try. 1000 seems reasonable on smaller sets, but will break on bigger sets (hence this issue).

Reply:
yeah that makes sense, i like slippage factor better tbh.

Reply:
I've made slippage a parameter.

Reply:
champ! thanks @sksamuel